plugins/modules/zabbix_usergroup.py

```diff
@@ -0,0 +1,597 @@
+            usergroup: added user group
+        """
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            parameters = self._construct_parameters(**kwargs)
+            usergroup = self._zapi.usergroup.create(parameters)
+            return usergroup['usrgrpids'][0]
+        except Exception as e:
+            self._module.fail_json(msg='Failed to create user group "%s": %s' % (kwargs['name'], e))
+
+    def delete(self, usrgrpid):
+        """Delete user group.
+
+        Parameters:
+            usrgrpid: User group id.
+
+        Returns:
+            usergroup: deleted user group
+        """
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            else:
+                return self._zapi.usergroup.delete([usrgrpid])
+        except Exception as e:
+            self._module.fail_json(msg='Failed to delete user group "%s": %s' % (usrgrpid, e))
+
+
+def compare_lists(l1, l2, diff_dict):
+    """
+    Compares l1 and l2 lists and adds the items that are different
+    to the diff_dict dictionary.
+    Used in recursion with compare_dictionaries() function.
+
+    Parameters:
+        l1: first list to compare
+        l2: second list to compare
+        diff_dict: dictionary to store the difference
+
+    Returns:
+        dict: items that are different
+    """
+    if len(l1) != len(l2):
+        diff_dict.append(l1)
+        return diff_dict
+    for i, item in enumerate(l1):
+        if isinstance(item, dict):
+            diff_dict.insert(i, {})
+            diff_dict[i] = compare_dictionaries(item, l2[i], diff_dict[i])
+        else:
+            if item != l2[i]:
+                diff_dict.append(item)
+    while {} in diff_dict:
+        diff_dict.remove({})
+    return diff_dict
+
+
+def compare_dictionaries(d1, d2, diff_dict):
+    """
+    Compares d1 and d2 dictionaries and adds the items that are different
+    to the diff_dict dictionary.
+    Used in recursion with compare_lists() function.
+
+    Parameters:
+        d1: first dictionary to compare
+        d2: second dictionary to compare
+        diff_dict: dictionary to store the difference
+
+    Returns:
+        dict: items that are different
+    """
+    for k, v in d1.items():
+        if k not in d2:
+            diff_dict[k] = v
+            continue
+        if isinstance(v, dict):
+            diff_dict[k] = {}
+            compare_dictionaries(v, d2[k], diff_dict[k])
+            if diff_dict[k] == {}:
+                del diff_dict[k]
+            else:
+                diff_dict[k] = v
+        elif isinstance(v, list):
+            diff_dict[k] = []
+            compare_lists(v, d2[k], diff_dict[k])
+            if diff_dict[k] == []:
+                del diff_dict[k]
+            else:
+                diff_dict[k] = v
+        else:
+            if v != d2[k]:
+                diff_dict[k] = v
+    return diff_dict
```

Comment:
can be added to module_utils/helpers if you want

Reply:
yap, was unsure to move, but yes, will do it. Looks better to cluster the parts in module_utils/helpers if they are used in more then one module.